expr/interval_compound_literal.go

```diff
@@ -0,0 +1,246 @@
+package expr
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/substrait-io/substrait-go/proto"
+	"github.com/substrait-io/substrait-go/types"
+)
+
+// IntervalCompoundLiteral creates an interval compound literal
+type IntervalCompoundLiteral struct {
+	years       int32
+	months      int32
+	days        int32
+	seconds     int32
+	subSeconds  int64
+	precision   types.TimePrecision
+	nullability types.Nullability
+}
```

Comment:
since all of the members are private, how does a user actually view or get the value in this literal? As written, it's impossible for any library to do anything with this literal except convert it to and from the protobuf object which makes it sort of useless unless we either make the members public or create accessor methods. 

This is why I was using the protobuf object directly, so I didn't have to duplicate all of the code to create the accessor methods

Reply:
@anshuldata, my request was to make the fields and structure public. No reason to hide.

Reply:
Also suggest we name the precision field  as subsecond precision or similar. Make it clear that only relates to that field...

Reply:
Fixed both
1. Made fields name Public
2. Renamed "precision" to "SubSecondPrecision"